### PR TITLE
Bring up adaptive navigation for reply study

### PR DESCRIPTION
**Pull request by Renzo-Olivares**
[Material spec](https://material.io/design/material-studies/reply.html#product-architecture) outlines the navigation experience for reply as follows:
Desktop: Navigation Drawer
Tablet: Navigation Rail
Mobile: BottomAppBar

This PR introduces a [ResponsiveWidget](https://medium.com/flutter-community/making-cross-platform-flutter-landing-page-responsive-7fffe0655970) class that uses a LayoutBuilder to return the appropriate navigation layout based on screen size given by MediaQueries. 

adaptive-nav

### CODE REVIEW

File: lib/studies/reply/inbox.dart [hunk 1]
```diff
@@ -1,12 +1,166 @@
...
+      ),
+      mobileScreen: _BuildMobileNav(
+        selectedIndex: _selectedIndex,
+        destinations: _navigationItems,
+        onItemTapped: _onItemTapped,
+      ),
+    );
+  }
+
+  void _onItemTapped(int index) {
```
Comment (clocksmith) [inline lib/studies/reply/inbox.dart]: _onItemSelected works better with a mix of taps and clicks

Comment (clocksmith) [reply lib/studies/reply/inbox.dart]: maybe onDestinationSelected?

File: lib/studies/reply/responsive_widget.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+class ResponsiveWidget extends StatelessWidget {
+  final Widget desktopScreen;
+  final Widget tabletScreen;
+  final Widget mobileScreen;
+
+  ResponsiveWidget(
+      {Key key, this.desktopScreen, this.tabletScreen, this.mobileScreen})
+      : super(key: key);
+
+  static bool isDesktop(BuildContext context) {
```
Comment (clocksmith) [inline lib/studies/reply/responsive_widget.dart:13]: Make a TODO note here to re-use the code in adaptive.dart once its finalized

**Review by clocksmith — APPROVED** [review]

File: lib/studies/reply/inbox.dart [hunk 3]
```diff
@@ -1,12 +1,166 @@
...
+  @override
+  void initState() {
+    super.initState();
+    _isExtended = widget.extended;
+  }
+
+  @override
+  void didUpdateWidget(_BuildDesktopNav oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.extended != widget.extended) _isExtended = widget.extended;
```
Comment (rami-a) [inline lib/studies/reply/inbox.dart]: Nit: I prefer: 

```suggestion
    if (oldWidget.extended != widget.extended) {
      _isExtended = widget.extended;
    }
```

**Review by rami-a — APPROVED** [review]

File: lib/studies/reply/inbox.dart (hunk 1 above)
Comment (Renzo-Olivares) [reply lib/studies/reply/inbox.dart]: I'll try that

**Pull request merged**